tests/scheduler/test_utils.py

```diff
@@ -3,10 +3,78 @@
 from rubin_sim.data import get_data_dir
 from rubin_sim.scheduler.utils import season_calc
 from rubin_sim.scheduler.model_observatory import ModelObservatory
-from rubin_sim.scheduler.utils import run_info_table
+from rubin_sim.scheduler.utils import run_info_table, restore_scheduler
+from rubin_sim.scheduler.example import example_scheduler
+from rubin_sim.scheduler import sim_runner
+import numpy as np
 
 
-class TestFeatures(unittest.TestCase):
+class TestUtils(unittest.TestCase):
+    @unittest.skipUnless(
+        os.path.isfile(os.path.join(get_data_dir(), "maps/DustMaps/dust_nside_32.npz")),
+        "Test data not available.",
+    )
+    def test_restore(self):
+        """Test we can restore a scheduler properly"""
+        # MJD set so it's in test data range
+        mjd_start = 60110.0
+        n_visit_limit = 3000
+
+        scheduler = example_scheduler(mjd_start=mjd_start)
+
+        mo = ModelObservatory(mjd_start=mjd_start)
+        mo, scheduler, observations = sim_runner(
+            mo,
+            scheduler,
+            survey_length=30.0,
+            verbose=False,
+            filename=None,
+            n_visit_limit=n_visit_limit,
+        )
+
+        # Won't be exact if we restart in the middle of a blob sequence, so need to restart on a new night
```

Comment:
Can you clarify this? Is it just because during a blob, the queue is generated all at once and not based on the actual conditions in the middle of the blob? (and does this mean we should mark "beginning of blobs"?)

Reply:
Exactly, we can't reconstruct the queue (or any scheduled observations that were generated earlier in the night).

Reply:
It's usually easy to pick out the start of blob sequences since they are tagged with "a" or "b".